src/rgw/rgw_sal_motr.h

```diff
@@ -1002,4 +1002,77 @@ class MotrStore : public Store {
     MotrMetaCache* get_bucket_inst_cache() {return bucket_inst_cache;}
 };
 
+struct obj_time_weight {
```

Comment:
This struct is available in `rgw_rados.cc, line number: 3674` file, but since it is a `.cc` file we can not import it directly, that is why I duplicated it in `rgw_sal_motr.h`.
Please provide input if we can avoid this.
@bkirunge7  @sachinpunadikar @sjain09  @jjxsg

Reply:
I think it's not defined with reusability considerations as it is not used anywhere else apart from `rgw_rados.cc`
So, if you want to use it, you have to redefine it in motr sal. Just check if you need all functions/attributes from `obj_time_weight` struct for our usecase. If not, then we can skip those.

Reply:
I'd suggest opening this PR against Ceph repo and ask Ceph devs' opinion about it.

Reply:
can we take their reviews on this PR itself?? 
Since this branch has lots of earlier commits that are not there in ceph repo, it will show a lot of changes there.

Reply:
Well, I'm not sure if they will - why would they? Of course, they would ask for PR against the Ceph repo. And yes - all the earlier commits this PR depends on should be in that PR as well...

Reply:
Hi, so sorry, got compiled for me as well. I must have got and resolved it with some other issue.
Reverted to `enum {OBJ_INSTANCE_LEN = 32};`.